### PR TITLE
Drop support for legacy kokkos target

### PR DESCRIPTION
**Pull request by dalg24**
Related to #2283 

It is more controversial as it might break user code but I think it is time to do it.

### CODE REVIEW

File: cmake/kokkos_install.cmake [hunk 1]
```diff
@@ -25,25 +25,11 @@ WRITE_BASIC_PACKAGE_VERSION_FILE("${Kokkos_BINARY_DIR}/KokkosConfigVersion.cmake
...
-# location that will not be discovered by CMake when Kokkos install prefix is
-# added to CMAKE_PREFIX_PATH in user code.
-install(FILES
-  "${Kokkos_BINARY_DIR}/KokkosConfig.cmake"
-  "${Kokkos_BINARY_DIR}/KokkosConfigVersion.cmake"
-  DESTINATION ${wrong_place})
-install(EXPORT KokkosTargets DESTINATION ${wrong_place})
-endif()
+  DESTINATION lib/cmake/Kokkos)
+install(EXPORT KokkosTargets NAMESPACE Kokkos:: DESTINATION lib/cmake/Kokkos)
```
Comment (junghans) [inline cmake/kokkos_install.cmake]:
```suggestion
install(EXPORT KokkosTargets NAMESPACE Kokkos:: DESTINATION ${INSTALL_LIB_DIR}/cmake/Kokkos)
```

File: cmake/kokkos_install.cmake [hunk 2]
```diff
@@ -25,25 +25,11 @@ WRITE_BASIC_PACKAGE_VERSION_FILE("${Kokkos_BINARY_DIR}/KokkosConfigVersion.cmake
...
-# For backward compatibility, export legacy target (not namespaced) to the old
-# location that will not be discovered by CMake when Kokkos install prefix is
-# added to CMAKE_PREFIX_PATH in user code.
-install(FILES
-  "${Kokkos_BINARY_DIR}/KokkosConfig.cmake"
-  "${Kokkos_BINARY_DIR}/KokkosConfigVersion.cmake"
-  DESTINATION ${wrong_place})
-install(EXPORT KokkosTargets DESTINATION ${wrong_place})
-endif()
+  DESTINATION lib/cmake/Kokkos)
```
Comment (junghans) [inline cmake/kokkos_install.cmake]:
```suggestion
  DESTINATION ${INSTALL_LIB_DIR}/cmake/Kokkos)
```

**Comment by dalg24** [comment]
> Actually it might be better drop the `INSTALL_{LIB,BIN,INCLUDE}_DIR` and use `GNUInstallDirs` instead.

I agree with you but I want to tackle this later.  This PR is only about forcing users to use the new `Kokkos::kokkos` target.

**Comment by dalg24** [comment]
@jjwilke please ignore https://github.com/kokkos/kokkos/pull/2287#pullrequestreview-284946089
and review the PR as is

**Comment by junghans** [comment]
> This could be a fairly dangerous change to make for users. We might be asking for trouble if we install the config files in different locations - unless `Kokkos_ROOT` is usable by everyone. I might delay this until we bump to 3.12... for which I believe the default behavior of CMP0074 is different?

As long as you don't mixed architectures the files will either be in `lib64` or `lib` and CMake usually is looking for configuration files in both. The only problem I see is if you want to overwrite an existing installation, but that is a problem with all kind of renamed. We could add an `uninstall` target to make a bit easier for the users. If backwards compatibility is a big issue let's make the libdir configurable but  leave the default with `lib` (even on 64-bit plattforms) for a while.

**Pull request merged**